### PR TITLE
Auto detect text file charsets. Partial implementation of #638

### PR DESCRIPTION
**Pull request by lwhite1**
Thanks for contributing.

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://developercertificate.org)

## Description

Detects encodings for input text files when possible, and when no encoding is specifically provided as an argument. If the detection confidence is weak, it falls back on the system default charset.

This implementation adds a new dependency to Core, which is undesirable, but getting data into tablesaw correctly is arguably its most important function so I think it's worth it.

## Testing

Did you add a unit test?
No but there are numerous tests that exercise this functionality

### CODE REVIEW

File: aggregate/pom.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
   <parent>
     <groupId>tech.tablesaw</groupId>
     <artifactId>tablesaw-parent</artifactId>
-    <version>0.34.3-SNAPSHOT</version>
+    <version>0.35.1-SNAPSHOT</version>
```
Comment (benmccann) [inline aggregate/pom.xml]: This file shouldn't be edited. I'm afraid you probably pulled while I was trying to cut a release earlier today, which failed and was rolled back

Comment (lwhite1) [reply aggregate/pom.xml]: ah  didn't realize it was rolled back.  I will unwind this tomorrow.

Comment (benmccann) [reply aggregate/pom.xml]: Unfortunately Maven is a bit frustrating in this regard. It'll commit the version bump and push it to Github early in the release process. If it fails anywhere in the process it's already pushed the commit despite the release failing

Comment (benmccann) [reply aggregate/pom.xml]: It looks like I got the release out (it takes an hour to show up, so I'll verify later). Unfortunately I had to do a force push so this PR is still showing a diff. Not sure the best way to solve. You might have to open a new PR?

Comment (lwhite1) [reply aggregate/pom.xml]: no problem. Thanks for getting it cleared-up.

**Comment by lwhite1** [comment]
replaced with a cleaner PR

**Pull request closed without merging**